include/nlohmann/detail/iterators/iteration_proxy.hpp

```diff
@@ -110,16 +111,45 @@ template<typename IteratorType> class iteration_proxy
         : container(cont) {}
 
     /// return iterator begin (needed for range-based for)
-    iteration_proxy_internal begin() noexcept
+    iteration_proxy_internal<IteratorType> begin() noexcept
     {
-        return iteration_proxy_internal(container.begin());
+        return iteration_proxy_internal<IteratorType>(container.begin());
     }
 
     /// return iterator end (needed for range-based for)
-    iteration_proxy_internal end() noexcept
+    iteration_proxy_internal<IteratorType> end() noexcept
     {
-        return iteration_proxy_internal(container.end());
+        return iteration_proxy_internal<IteratorType>(container.end());
     }
 };
+// structured bindings support
+template <std::size_t N, typename IteratorType, typename std::enable_if<N == 0, int>::type = 0>
+auto get(const nlohmann::detail::iteration_proxy_internal<IteratorType>& i) -> decltype(i.key())
+{
+    return i.key();
+}
+
+template <std::size_t N, typename IteratorType, typename std::enable_if<N == 1, int>::type = 0>
+auto get(const nlohmann::detail::iteration_proxy_internal<IteratorType>& i) -> decltype(i.value())
+{
+    return i.value();
+}
 }  // namespace detail
 }  // namespace nlohmann
+
```

Comment:
Could you please add a comment why you are adding code to the `std` namespace?

Reply:
Would something like the following be fine?
```
// The Addition to the STD Namespace is required to add
// Structured Bindings Support to the iteration_proxy_value class
// For further reference see https://blog.tartanllama.xyz/structured-bindings/
```
Or would a TartanLlama link be inappropriate and I should add the less helpful cppreference link
https://en.cppreference.com/w/cpp/language/structured_binding

Reply:
The comment is fine. Maybe also add a link to this PR as reference.

Reply:
@pratikpc If you add this, I would be ready to merge.

Reply:
@nlohmann I made the required modifications